h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/exec/XGBoostHttpClient.java

```diff
@@ -0,0 +1,86 @@
+package hex.tree.xgboost.exec;
+
+import hex.genmodel.utils.IOUtils;
+import hex.schemas.XGBoostExecReqV3;
+import hex.schemas.XGBoostExecRespV3;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.apache.log4j.Logger;
+import water.Key;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.http.HttpHeaders.CONTENT_TYPE;
+import static water.util.HttpResponseStatus.OK;
+
+public class XGBoostHttpClient {
+
+    private static final Logger LOG = Logger.getLogger(XGBoostHttpClient.class);
+
+    private final String baseUri;
+
+    interface ResponseTransformer<T> {
+        T transform(HttpEntity e) throws IOException;
+    }
+
+    private static final ResponseTransformer<byte[]> ByteArrayResponseTransformer = (e) -> {
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
```

Comment:
`IOUtils.copyStream` does not close the output stream (this patter can actually be seen in H2O). We'll need to close it.

Reply:
close on BOS is a NOOP, but adding for consistency